src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage.cs

```diff
@@ -233,63 +246,84 @@ private PooledConnection GetPooledConnection()
+
+            // Set "synchronous" mode to "normal" instead of default "full" to reduce the amount of buffer flushing syscalls,
+            // significantly reducing both the blocked time and the amount of context switches.
+            // When coupled with WAL, this (according to https://sqlite.org/pragma.html#pragma_synchronous and 
+            // https://www.sqlite.org/wal.html#performance_considerations) is unlikely to significantly affect durability,
+            // while significantly increasing performance, because buffer flushing is done for each checkpoint, instead of each
+            // transaction. While some writes can be lost, they are never reordered, and higher layers will recover from that.
+            connection.ExecuteCommand("pragma synchronous=normal", throwOnError: false);
+
+            // First, create all string tables in the main on-disk db.  These tables
+            // don't need to be in the write-cache as all string looks go to/from the
+            // main db.  This isn't a perf problem as we write the strings in bulk, 
+            // so there's no need for a write caching layer.  This also keeps consistency
+            // totally clear as there's only one source of truth.
+            connection.ExecuteCommand(
+$@"create table if not exists ""{MainDBName}.{StringInfoTableName}"" (
     ""{DataIdColumnName}"" integer primary key autoincrement not null,
     ""{DataColumnName}"" varchar)");
 
-                // Ensure that the string-info table's 'Value' column is defined to be 'unique'.
-                // We don't allow duplicate strings in this table.
-                connection.ExecuteCommand(
-$@"create unique index if not exists ""{StringInfoTableName}_{DataColumnName}"" on ""{StringInfoTableName}""(""{DataColumnName}"")");
+            // Ensure that the string-info table's 'Value' column is defined to be 'unique'.
+            // We don't allow duplicate strings in this table.
+            connection.ExecuteCommand(
+$@"create unique index if not exists ""{MainDBName}.{StringInfoTableName}_{DataColumnName}"" on ""{StringInfoTableName}""(""{DataColumnName}"")");
+
+            // Now create the individual tables for the solution/project/document info.
+            CreateTables(connection, MainDBName);
 
+            // Also get the known set of string-to-id mappings we already have in the DB.
+            // Do this in one batch if possible.
+            var fetched = TryFetchStringTableFromMainDB(connection);
+
+            // If we weren't able to retrieve the entire string table in one batch,
+            // attempt to retrieve it for each 
+            var fetchStringTable = !fetched;
+
+            // Try to bulk populate all the IDs we'll need for strings/projects/documents.
+            // Bulk population is much faster than trying to do everything individually.
+            BulkPopulateIds(connection, solution, fetchStringTable);
```

Comment:
note the logic from `var fetched` to here is the same as before.  it's just that the lines above it were extracted out into `CreateTables`